clients/multiversx/client_test.go

```diff
@@ -1103,6 +1109,42 @@ func TestClient_CheckClientAvailability(t *testing.T) {
 	})
 }
 
+func TestClient_GetBatchSCMetadata(t *testing.T) {
+	t.Parallel()
+
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockClientArgs()
+		args.Proxy = &interactors.ProxyStub{
+			FilterLogsCalled: func(ctx context.Context, filter *core.FilterQuery) ([]*transaction.Events, error) {
+				return []*transaction.Events{{Identifier: "event0"}, {Identifier: "event1"}}, nil
```

Comment:
I would complicate a little bit this test by setting the deposits from the test batch to hold 2 non-zero values for the DepositBlockNonce. Along with setting some values for the eventsBlockRangeFrom and eventsBlockRangeTo to check that the additon is done.
Also, I would add more tests just to check some extreme situations: a depositBlockNumer of 37, for example and a eventsBlockRangeFrom of -38. (In this extreme case the value should not overflow - production code might need adjustments)